### PR TITLE
Improved input validation

### LINKED ISSUE #1400: Use new rlang check_ helpers
Make sure to do #1340 first.

### PR DESCRIPTION
**Pull request by hadley**
Using rlang's new `check_` helpers. Fixes #1400

### CODE REVIEW

File: R/nest-legacy.R [hunk 1]
```diff
@@ -147,16 +147,16 @@ unnest_legacy.data.frame <- function(data, ..., .drop = NA, .id = NULL,
...
   nest_types <- split.default(nested, types)
   if (length(nest_types$mixed) > 0) {
-    probs <- paste(names(nest_types$mixed), collapse = ",")
-    abort(glue(
-      "Each column must either be a list of vectors or a list of ",
-      "data frames [{probs}]"
+    probs <- names(nest_types$mixed)
+    cli::cli_abort(c(
+      "Each column must either be a list of vectors or a list of data frames.",
+      i = "Problems in: {.var probs}"
```
Comment (DavisVaughan) [inline R/nest-legacy.R:159]: `"Problems in: {.var {probs}}"`

I think you wanted this? i.e. interpolate `probs`? saw it in one of the error snapshots

File: tests/testthat/_snaps/pivot-wide.md [hunk 2]
```diff
@@ -57,7 +57,7 @@
     Output
       <error/rlang_error>
       Error in `build_wider_spec()`:
-      ! `values_from` must select at least one column.
+      ! Must select at least one item.
```
Comment (DavisVaughan) [inline tests/testthat/_snaps/pivot-wide.md:60]: These used to report `names_from` vs `values_from` as the place where the problem came from, which was nice

File: tests/testthat/_snaps/pivot-wide.md [hunk 3]
```diff
@@ -162,9 +162,9 @@
       * Use `values_fn = {summary_fun}` to summarise duplicates.
       * Use the following dplyr code to identify duplicates.
         {data} %>%
-          dplyr::group_by(a, key) %>%
-          dplyr::summarise(n = dplyr::n(), .groups = "drop") %>%
-          dplyr::filter(n > 1L)
+        dplyr::group_by(a, key) %>%
+        dplyr::summarise(n = dplyr::n(), .groups = "drop") %>%
+        dplyr::filter(n > 1L)
```
Comment (DavisVaughan) [inline tests/testthat/_snaps/pivot-wide.md:167]: I think the tabbing is off? i.e. `{data}` shouldn't be aligned with the following lines right?

File: tests/testthat/_snaps/uncount.md [hunk 4]
```diff
@@ -1,9 +1,23 @@
...
+    Code
+      uncount(df, x, .remove = 1)
+    Condition
+      Error in `uncount()`:
+      ! `.remove` must be `TRUE` or `FALSE`, not the number 1.
+    Code
+      uncount(df, x, .id = "")
+    Condition
+      Error in `uncount()`:
+      ! `.id` must be a valid name or `NULL`, not the empty string "".
```
Comment (DavisVaughan) [inline tests/testthat/_snaps/uncount.md:22]: I noticed in stringr error messages you backtick the `""` but we didn't do that in the rlang check functions

File: tests/testthat/_snaps/pivot-wide.md (hunk 2 above)
Comment (hadley) [reply tests/testthat/_snaps/pivot-wide.md:60]: Issue: https://github.com/r-lib/tidyselect/issues/327

File: tests/testthat/_snaps/pivot-wide.md (hunk 3 above)
Comment (hadley) [reply tests/testthat/_snaps/pivot-wide.md:167]: Right, but I gave up getting cli to not re-indent.

Comment (DavisVaughan) [reply tests/testthat/_snaps/pivot-wide.md:167]: Oh that is unfortunate

**Pull request merged**